lib/src/input_validation/validate_function.dart

```diff
@@ -40,23 +39,151 @@ class ValidateFunction {
     List<String> inputString = _sanitizeInput(input);
     State currentState = StartState(0, false);
 
-    for(int i=0;(i<inputString.length && valid);i++) {
+    for (int i = 0; (i < inputString.length && valid); i++) {
       String token = _sanitizeToken(inputString[i]);
 
-      if(operators.contains(token) || Pattern.validOperand.hasMatch(token)) {  // numbers or operands
+      if (operators.contains(token) ||
+          Pattern.validOperand.hasMatch(token)) { // numbers or operands
         currentState = currentState.getNextState(token);
-      } else if(multiParamFunctions.contains(token)){
+      } else if (multiParamFunctions.contains(token)) {
         currentState.multiParam = true;
-      } else if(!validFunctions.contains(token)) {
-        valid =  false;
+      } else if (!validFunctions.contains(token)) {
+        valid = false;
       }
-
-      if(currentState is ErrorState) {
+      if (currentState is ErrorState) {
         valid = false;
       }
     }
 
     return valid;
   }
 
+  bool testMatrixFunction(String expression){
```

Comment:
lets move this to a separate class, its pretty large and does something pretty separate

Reply:
will do

Reply:
done